XXH3 and Windows ARM builds

I have a problem with using `xxh3.h` on Widows while building for an ARM architecture.
The problem is that `__emulu` is an x64/x86 only windows internal macro ([MS Docs](https://docs.microsoft.com/en-us/cpp/intrinsics/emul-emulu?view=vs-2019)).
```
/* U64 XXH_mult32to64(U32 a, U64 b) { return (U64)a * (U64)b; } */
#ifdef _MSC_VER
#   include <intrin.h>
    /* MSVC doesn't do a good job with the mull detection. */
#   define XXH_mult32to64 __emulu
#else
#   define XXH_mult32to64(x, y) ((U64)((x) & 0xFFFFFFFF) * (U64)((y) & 0xFFFFFFFF))
#endif
```
Where it's defined only checks for an `_MSC_VER` before using it. It would be very helpful if this could check architecture as well. Say either by not including it when ARM or only using it on x64/x86 processors. Thanks.

When I run these steps using CMake and Visual Studio 2019:

1. `cd cmake_unofficial`
2. `mkdir build`
3. `cd build`
4. `cmake -G "Visual Studio 16 2019" -A ARM ..`
5. `cmake --build .`

I get this error:
```
xxhash.obj : error LNK2019: unresolved external symbol __emulu referenced in function XXH3_mul128_fold64 [C:\Users\steve\Do
cuments\xxHash\cmake_unofficial\build\xxhash.vcxproj]
C:\Users\steve\Documents\xxHash\cmake_unofficial\build\Debug\xxhash.dll : fatal error LNK1120: 1 unresolved externals [C:\U
sers\steve\Documents\xxHash\cmake_unofficial\build\xxhash.vcxproj]
```

I don't have much experience with appveyor.yml, but following the existing pattern, I think this will let you generate and run for ARM:
```
- if "%PLATFORM%"=="visual_arm" (
      cd cmake_unofficial &&
      cmake . -DCMAKE_BUILD_TYPE=Release -A ARM &&
      cmake --build . --config Release
  )
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -5,6 +5,8 @@ environment:
     PLATFORM: "visual_x64"
   - COMPILER: "visual"
     PLATFORM: "visual_x86"
+  - COMPILER: "visual"
+    PLATFORM: "visual_arm"
   - COMPILER: "gcc"
     PLATFORM: "mingw64"
   - COMPILER: "gcc"
@@ -51,6 +53,11 @@ build_script:
   - if "%PLATFORM%"=="visual_x86" (
       cd cmake_unofficial &&
       cmake . -DCMAKE_BUILD_TYPE=Release &&
+      cmake --build . --config Release
+    )
+  - if "%PLATFORM%"=="visual_arm" (
+      cd cmake_unofficial &&
+      cmake . -DCMAKE_BUILD_TYPE=Release -A ARM &&
       cmake --build . --config Release
     )
 

--- a/xxh3.h
+++ b/xxh3.h
@@ -121,12 +121,11 @@
 #endif
 
 /* U64 XXH_mult32to64(U32 a, U64 b) { return (U64)a * (U64)b; } */
-#ifdef _MSC_VER
-#   include <intrin.h>
-    /* MSVC doesn't do a good job with the mull detection. */
-#   define XXH_mult32to64 __emulu
+#if defined(_MSC_VER) && defined(_M_IX86)
+#    include <intrin.h>
+#    define XXH_mult32to64 __emulu
 #else
-#   define XXH_mult32to64(x, y) ((U64)((x) & 0xFFFFFFFF) * (U64)((y) & 0xFFFFFFFF))
+#    define XXH_mult32to64(x, y) ((U64)((x) & 0xFFFFFFFF) * (U64)((y) & 0xFFFFFFFF))
 #endif
 
 /* VSX stuff */

